tests/browser/async_mainloop.cpp

```diff
@@ -10,7 +10,7 @@
 extern "C" {
 
 void EMSCRIPTEN_KEEPALIVE finish(int result) {
-  REPORT_RESULT(result);
+  emscripten_force_exit(result);
```

Comment:
Can you instead use `assert(result ==121)` here and then `emscripten_force_exit(0);` which should allow your to completely remove the `assert_returncode` argument to `btest_exit`.

I've been leaning towards that style as much as possible and I think it has several advantages.